### ISSUE TITLE
Test failure in testNondeterministicTask.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#356]
I see this test failure pretty frequently (in `stress_tests.py`). However, I haven't been able to reproduce it locally.

```
======================================================================
FAIL: testNondeterministicTask (__main__.ReconstructionTestsMultinode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/stress_tests.py", line 357, in testNondeterministicTask
    self.assertTrue(len(errors) >= num_objects / 2)
AssertionError: False is not true

----------------------------------------------------------------------
Ran 13 tests in 413.376s
```

@stephanie-wang

**Comment by robertnishihara** [comment]
We should also make that test less verbose... since it prints a lot of copies of

```
[INFO] (/Users/rkn/Workspace/ray/src/plasma/eviction_policy.cc:181) There is not enough space to create this object, so evicting 25 objects to free up 50029800 bytes.
[WARN] (/Users/rkn/Workspace/ray/src/common/state/redis.cc:232) Found objects with different value but same object ID, most likely because a nondeterministic task was executed twice, either for reconstruction or for speculation.
A nondeterministic task was reexecuted.

You can inspect errors by running

    ray.error_info()

If this driver is hanging, start a new one with

    ray.init(redis_address="127.0.0.1:51441")
```

**Comment by robertnishihara** [comment]
It's possible that the problem is that we aren't waiting long enough and we could fix this with something like #357.

**Comment by robertnishihara** [comment]
Note that the verbosity is addressed (at least in part) by #377.

**Comment by stephanie-wang** [comment]
The problem turned out to be that we were waiting for too many errors to appear in `ReconstructionTestsMultinode`. In a single-node environment, we call `ray.get` on the objects sequentially, so each object gets evicted and reconstructed exactly once. Half the objects are created by nondeterministic tasks, so we should get exactly that many errors by the end. In a multinode setting, however, depending on how tasks are scheduled, some objects may not get evicted at all, while others may get evicted more than necessary. Then, we cannot guarantee that the number of errors will be exactly the number of nondeterministic tasks.

Fixed in #382.

**Comment by robertnishihara** [comment]
Closing because this should be fixed by #382.
